tests/plugin-parser/test-mlops-plugin-parser.cc

```diff
@@ -0,0 +1,397 @@
+}
+
+/**
+ * @brief Negative test case of pkgmgrinfo_pkginfo_get_res_version() failed.
+ */
+TEST_F (PkgMgrInfoMockTestFixture, test_n5)
+{
+  pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
+  std::string pkgid = "pkgid";
+  std::string appid = "appid";
+  GList *md_list = nullptr;
+
+  EXPECT_CALL (*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_destroy_pkginfo (_))
+      .WillRepeatedly (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_pkginfo (_, _))
+      .WillOnce (Return (PMINFO_R_OK));
+
+  char pkg_type_rpk[] = "rpk";
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_type (_, _))
+      .WillOnce (DoAll (SetArgPointee<1> (pkg_type_rpk),
+                        Return (PMINFO_R_OK)));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_root_path (_, _))
+      .WillOnce (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_res_type (_, _))
+      .WillOnce (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_res_version (_, _))
+      .WillOnce (Return (PMINFO_R_ERROR));
+
+  EXPECT_EQ (exec_plugin_parser_func ("PKGMGR_MDPARSER_PLUGIN_INSTALL",pkgid, appid, md_list), -1);
+
+  delete pkgMgrInfoMockInstance;
+}
+
+/**
+ * @brief Positive test case of sample rpk package install.
+ */
+TEST_F (PkgMgrInfoMockTestFixture, test_p1)
+{
+  pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
+  std::string pkgid = "pkgid";
+  std::string appid = "appid";
+  GList *md_list = nullptr;
+
+  EXPECT_CALL (*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_destroy_pkginfo (_))
+      .WillRepeatedly (Return (PMINFO_R_OK));
+
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_pkginfo (_, _))
+      .WillOnce (Return (PMINFO_R_OK));
+
+  char pkg_type_rpk[] = "rpk";
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_type (_, _))
+      .WillOnce (DoAll (SetArgPointee<1> (pkg_type_rpk),
+                        Return (PMINFO_R_OK)));
+
+  char root_path[] = "../tests/plugin-parser/test_rpk_sample";
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_root_path (_, _))
+      .WillOnce (DoAll (SetArgPointee<1> (root_path),
+                        Return (PMINFO_R_OK)));
+
+  char res_type[] = "sample-res-type";
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_res_type (_, _))
+      .WillOnce (DoAll (SetArgPointee<1> (res_type),
+                      Return (PMINFO_R_OK)));
+
+  char res_version[] = "1.5.0";
+  EXPECT_CALL(*pkgMgrInfoMockInstance, pkgmgrinfo_pkginfo_get_res_version (_, _))
+      .WillOnce (DoAll (SetArgPointee<1> (res_version),
+                      Return (PMINFO_R_OK)));
+
+  EXPECT_EQ (exec_plugin_parser_func ("PKGMGR_MDPARSER_PLUGIN_INSTALL",pkgid, appid, md_list), 0);
+
+  delete pkgMgrInfoMockInstance;
+}
+
+/**
+ * @brief Positive test case of tpk package.
+ */
+TEST_F (PkgMgrInfoMockTestFixture, test_tpk)
+{
+  pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
```

Comment:
Suggestion: create new instance in setup() and release it in down()
pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
delete pkgMgrInfoMockInstance;

Or, can we set the mock instance in main() function?
```
  try {
    pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
    result = RUN_ALL_TESTS ();
    delete pkgMgrInfoMockInstance;
  } catch (...) {
    g_warning ("catch `testing::internal::GoogleTestFailureException`");
  }
```

Reply:
And also, plz check enabling test daemon in main()
```
  try {
    // enable test daemon
    pkgMgrInfoMockInstance = new PkgMgrInfoMock ();
    result = RUN_ALL_TESTS ();
    delete pkgMgrInfoMockInstance;
    // disable test daemon
  } catch (...) {
    g_warning ("catch `testing::internal::GoogleTestFailureException`");
  }
```

Reply:
I let the test use static method `SetUpTestSuite` and `TearDownTestSuite` to be called only once. `PkgMgrInfoMock`, `GTestDBus`, and `GModule` are initialized and deleted only once.